shared/src/commonMain/kotlin/org/hyperskill/app/track_selection/presentation/TrackSelectionListFeature.kt

```diff
@@ -0,0 +1,87 @@
+package org.hyperskill.app.track_selection.presentation
+
+import org.hyperskill.app.analytic.domain.model.AnalyticEvent
+import org.hyperskill.app.track.domain.model.Track
+import org.hyperskill.app.track.domain.model.TrackWithProgress
+
+object TrackSelectionListFeature {
+    internal sealed interface State {
+        object Idle : State
+        object Loading : State
+        data class Content(
+            val tracks: List<TrackWithProgress>,
+            val selectedTrackId: Long?
+        ) : State
+        object NetworkError : State
+    }
+
+    sealed interface ViewState {
+        object Idle : ViewState
+        object Loading : ViewState
+        object Error : ViewState
+        data class Content(
+            val selectedTrack: Track?,
+            val tracks: List<Track>
```

Comment:
@ivan-magda 
Agree, we can just provide a properly sorted list of tracks, with the `isSelected` property in the Track model.

Reply:
I don't like idea of depending on order of tracks, it will be harder to support this logic in cases when we shall change ordering of tracks. I would like to leave this solution, but if you insist, I'll change it.

Reply:
> I don't like idea of depending on order of tracks, it will be harder to support this logic in cases when we shall change ordering of tracks.

@vladkash what kind of ordering do you mean? This is absolutely correct that shared returns a collection of sorted tracks by some business logic.

Reply:
@ivan-magda @XanderZhu share your opinion please, is it definetly required?

Reply:
@vladkash 
For sure, it is not strongly required.
But in my opinion, the option with a sorted list is better, because we do all the sorting on the shared side, not on the platform side.

Reply:
@vladkash please make changes:

`ViewState.Content`:
1.  Will only provide `tracks` property
2. Selected track always should be the first element of the `tracks` collection

`ViewState.Track`:
1. Add `isSelected` property